src/lib/migrations/routines/09-delete-empty-conversations.ts

```diff
@@ -0,0 +1,77 @@
+import type { Migration } from ".";
+import { collections } from "$lib/server/database";
+import { ObjectId } from "mongodb";
+import { logger } from "$lib/server/logger";
+
+const BATCH_SIZE = 1000;
+const DELETE_THRESHOLD_MS = 60 * 60 * 1000;
+
+export async function deleteConversations(
+	collections: typeof import("$lib/server/database").collections
+) {
+	let deleteCount = 0;
+	const { conversations, sessions } = collections;
+
+	const validSessionIdsArray = await sessions.distinct("sessionId");
```

Comment:
Not sure how it's going to work with such a large collection. Especially if you want to include the huge array in the query afterwards.

You may be better off running that check manually - eg get conversations without users 1000 by 1000 and then do `db.dessions.find({_id: {$in: ...}})` and see which Ids are missing, and delete those conversations

Reply:
Oh thanks, yes that would make more sense.